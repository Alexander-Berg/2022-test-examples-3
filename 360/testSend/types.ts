import { FEATURE_NAME } from './constants';

export interface ITestSendState {
    isActive: boolean;
    isLoading: boolean;
    error: string | null;
}

export interface IReduxState {
    [FEATURE_NAME]: ITestSendState;
}
