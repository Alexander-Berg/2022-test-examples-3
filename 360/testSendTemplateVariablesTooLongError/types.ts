export type TTestSendTemplateVariablesTooLongError = {
  (
    params: Readonly<{
      count: number;
    }>
  ): string;
};
