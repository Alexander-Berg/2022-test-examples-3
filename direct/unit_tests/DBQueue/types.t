#!/usr/bin/perl
use Direct::Modern;

=head1 NAME

DBQueue/types.t

=head1 DESCRIPTION

Проверяет формат YAML-файла etc/dbqueue-types.yaml

=cut

use Test::Deep;
use Test::More;
use YAML 'LoadFile';

use Yandex::DBQueue::Typemap;
use Yandex::Validate;

use Settings;

plan tests => 1;

my $DB_TYPE_LENGTH = 30;

sub check_id {
    my $id = shift; 
    state %IDS;

    if (!is_valid_id($id)) {
        return (0, "id is not valid number");
    }
    if ($IDS{$id}) {
        return (0, "id isn't unique");
    }

    $IDS{$id}++;

    return 1;
}

sub check_type {
    my $type = shift;
    state %TYPES;

    if ($type !~ qr/^\w+$/) {
        return (0, 'invalid type format, expected: ^\w+$');
    }
    if (length($type) > $DB_TYPE_LENGTH) {
        return (0, "type must not be longer than $DB_TYPE_LENGTH characters");
    }
    if ($TYPES{$type}) {
        return (0, "type isn't unique");
    }

    $TYPES{ $type }++;

    return 1;
}

cmp_deeply(
    YAML::LoadFile( $Yandex::DBQueue::Typemap::SOURCE_FILE_PATH ),
    array_each(
        {
            id => code(\&check_id),
            type => code(\&check_type),
        }
    ),
    "$Yandex::DBQueue::Typemap::SOURCE_FILE_PATH format ok"
);
