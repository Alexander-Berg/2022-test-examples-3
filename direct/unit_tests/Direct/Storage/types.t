#!/usr/bin/perl

=head1 DESCRIPTION

Тест проверяет соответствие между типами файлов, опсанными в Direct::Storage::Types
и полем type в таблицах mds_metadata (ppc и ppcdict)

=cut

use Direct::Modern;
use Test::More;
use Direct::Storage::Types;
use Settings;
use Path::Tiny;
use List::MoreUtils qw(uniq);

my %IGNORE_TYPES = map {$_ => 1} qw(
    sale_reports
);

=head2 get_types_from_schema

Распарсить (регуляркой) схему таблицы, вернуть содержимое enum

Параметры:
    $path - путь к файлу со схемой

Возвращает:
    хеш вида
        ( type => 1, type2 => 1 )

=cut

sub get_types_from_schema
{
    my $path = shift;
    my $text = path($path)->slurp();

    $text =~ s!\`!!g;
    my ($types) = ($text =~ /type \s+ enum \s* \((.+?)\) \s+ NOT \s+ NULL/gx);
    return
        map { $_ => 1 }
        map { s!\'!!gr }
        split /[, ]+/, $types;
}

my %types_ppc     = get_types_from_schema(path($Settings::ROOT, 'db_schema/ppc/mds_metadata.schema.sql'));
my %types_ppcdict = get_types_from_schema(path($Settings::ROOT, 'db_schema/ppcdict/mds_metadata.schema.sql'));

for my $type (keys %Direct::Storage::Types::MDS_FILE_TYPES) {
    if (delete $IGNORE_TYPES{$type}) {
        delete $types_ppcdict{$type};
        delete $types_ppc{$type};
    } elsif (mds_check_type_trait($type, 'empty_client_id')) {
        ok($types_ppcdict{$type} && !$types_ppc{$type}, "$type in schema for ppcdict.mds_metadata");
        delete $types_ppcdict{$type};
    }
    else {
        ok(!$types_ppcdict{$type} && $types_ppc{$type}, "$type in schema for ppc.mds_metadata");
        delete $types_ppc{$type};
    }
}

for my $missing (uniq ((keys %types_ppc), (keys %types_ppcdict))) {
    fail("missing type description for $missing (found in schema, but not in Direct::Storage::Types)");
}

my $unused_ignores = join(',', keys %IGNORE_TYPES);
is($unused_ignores, '', 'no unused IGNORE_TYPES');

done_testing();
