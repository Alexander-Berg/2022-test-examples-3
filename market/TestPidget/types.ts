export type Value = {
    fibonacciNumber: number;
};
