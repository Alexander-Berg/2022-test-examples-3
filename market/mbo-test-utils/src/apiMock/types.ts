export type PromiseCall<Response, Args extends any[]> = (...args: Args) => Promise<Response>;

export type MockedApiCall<T> = T extends PromiseCall<infer Response, infer Args> ? PromiseMock<Response, Args> : T;

export type MockedApi<T> = MockedApiApi & { [P in keyof T]: MockedApiCall<T[P]> };

export interface IRequestMock<Response, Args extends any[]> {
  args(): Args;
  resolve(response: Response): void;
  reject(error: any): void;
}

export interface PromiseMock<Response, Args extends any[]> extends jest.Mock<Promise<Response>, Args> {
  next(filter?: (...args: Args) => boolean | undefined): IRequestMock<Response, Args>;
  process(processor: (request: IRequestMock<Response, Args>) => void): void;
  activeRequests(): Args[];
}

export type MockedApiObject<T> = { [P in keyof T]: MockedApi<T[P]> & T[P] } & {
  allActiveRequests: () => { [P in keyof T]?: RequestInfo[] };
};

export interface CallInfo<Response, Args extends any[]> {
  args: Args;
  finalized: boolean;
  resolve: (value: Response) => void;
  reject: (error: any) => void;
}

export interface RequestInfo {
  name: string;
  requests: any[];
}

export interface MockPromiseExtension {
  beforeResolve?: (description: string, args: any) => void;
  afterResolve?: (description: string, args: any) => void;
  beforeReject?: (description: string, args: any) => void;
  afterReject?: (description: string, args: any) => void;
}

export interface MockedApiApi {
  activeRequests(): RequestInfo[];
}

interface IConfigController {
  getConfig<T>(): Promise<T>;
}

export interface IConfigApi {
  configController: IConfigController;
}
