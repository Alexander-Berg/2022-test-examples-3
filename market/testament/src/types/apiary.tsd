// eslint-disable-next-line max-classes-per-file
declare module '@yandex-market/apiary/client/runtime' {
    export default class Runtime {
        constructor(params: {strategy: unknown});

        run: () => void;
    }
}

declare module '@yandex-market/apiary/server/processor' {
    export default class Processor {
        static process(params: unknown): Promise<unknown>;
    }
}

declare module '@yandex-market/apiary-annex/jobs' {
    export function cookiesJob(params: unknown): void;
    export function statusJob(code: number, lock: boolean): void;
    export function titleJob(title: Promise<string>): void;
    export function metaJob(name: string, value: string): void;
    export function redirectJob(url: string): void;
}

declare module '@yandex-market/apiary' {
    import {Component, FC} from 'react';
    import {Action} from 'redux';
    import {ActionsObservable} from 'redux-observable';
    import {Observable} from 'rxjs';

    export type GenericAction = Action<string> & {
        payload?: unknown;
        meta?: {
            [key: string]: unknown;
        };
    };

    export type WidgetFacade<D, C> = {
        getData(): D;
        getCollections(): C;
    };

    export type Epic<A extends Action, S> = (
        action: ActionsObservable<A>,
        state: S,
    ) => Observable<Action>;

    export type WidgetEpic<A extends GenericAction, D, C> = Epic<
        A,
        WidgetFacade<D, C>
    >;

    export type Collections = {
        [key: string]: {
            [key: string]: unknown;
        };
    };
    export type State<TWidgetData, TCollections extends Collections> = {
        widgetData: TWidgetData;
        collections: TCollections;
    };

    export function connect(
        mapDataToProps: unknown,
        mapDispatchToProps: unknown,
    ): (comp: Component | FC) => Component<unknown>;
    export class Widget {
        static describe(params: unknown): void;
    }
}
