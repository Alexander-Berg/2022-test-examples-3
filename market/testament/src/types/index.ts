declare module 'asker' {
    export default class Request {
        done(err: Error | null, data: unknown): void;

        getResponseMetaBase(): {
            time: {network: number; total: number};
            retries: {used: number; limit: number};
        };
    }
}
