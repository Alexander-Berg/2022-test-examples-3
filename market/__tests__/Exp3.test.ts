import ApiMock from 'core-legacy/api/mock/ApiMock'
import createEnv from 'core-legacy/test-utils/create-env'
import serializeModel from 'core-legacy/test-utils/serialize-model'
import Exp3 from '../index'
import {VendorApi} from 'core-legacy/types/restyped'

describe('Exp3', () => {
  const apiMock = new ApiMock<VendorApi>()

  apiMock.mockPost('/api/configs' as any, {
    items: [
      {name: 'restapp_logs', value: {requestInterval: 1000}},
      {name: 'restapp_place_edit', value: {address_editable: false, name_editable: false, enabled: false}},
      {name: 'restapp_bundle_update_interval', value: {interval: 300}},
      {name: 'restapp_communications', value: {enabled: false}},
      {name: 'restapp_menu_sufficient_photo_widget', value: {enabled: true, sufficient_photo_percent: 75}},
      {name: 'restapp_discounts', value: {enabled: false, roles: ['ROLE_MANAGER']}},
      {
        name: 'restapp_push_notifications',
        value: {
          history_polling_socket_connected: 360,
          history_polling_socket_disconnected: 90,
          orders_polling_socket_connected: 120,
          orders_polling_socket_disconnected: 30
        }
      },
      {name: 'restapp_common', value: {configsPollInterval: 600000}},
    ],
    version: 361418
  })

  apiMock.mockPost('/4.0/restapp-front/eats/v1/ab/experiments' as any, {
    version: 333489,
    items: [{name: 'restapp_header_help_button', value: {enabled: true}}]
  })

  test('Initialization', async () => {
    const exp3 = Exp3.create({}, createEnv({api: apiMock}))

    expect(serializeModel(exp3)).toMatchSnapshot()

    await exp3.initialize()

    expect(serializeModel(exp3)).toMatchSnapshot()
  })

  test('Initialization reject on request fail', () => {
    apiMock.mock.onPost('/api/configs').reply(500, {})

    const exp3 = Exp3.create({}, createEnv({api: apiMock}))

    // tslint:disable-next-line:no-floating-promises
    expect(exp3.initialize()).rejects.toBeDefined()
  })
})
