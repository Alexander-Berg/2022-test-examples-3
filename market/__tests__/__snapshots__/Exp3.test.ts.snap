// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exp3 Initialization 1`] = `
Object {
  "configs": Object {},
  "experiments": Object {},
}
`;

exports[`Exp3 Initialization 2`] = `
Object {
  "configs": Object {
    "restapp_bundle_update_interval": Object {
      "interval": 300,
    },
    "restapp_common": Object {
      "configsPollInterval": 600000,
    },
    "restapp_communications": Object {
      "enabled": false,
    },
    "restapp_discounts": Object {
      "enabled": false,
      "roles": Array [
        "ROLE_MANAGER",
      ],
    },
    "restapp_logs": Object {
      "requestInterval": 1000,
    },
    "restapp_menu_sufficient_photo_widget": Object {
      "enabled": true,
      "sufficient_photo_percent": 75,
    },
    "restapp_place_edit": Object {
      "address_editable": false,
      "enabled": false,
      "name_editable": false,
    },
    "restapp_push_notifications": Object {
      "history_polling_socket_connected": 360,
      "history_polling_socket_disconnected": 90,
      "orders_polling_socket_connected": 120,
      "orders_polling_socket_disconnected": 30,
    },
    "restapp_site_constructor": Object {
      "constructorBaseUrl": "https://edc-develop.lc-internal.yandex.ru",
      "enabled": false,
    },
  },
  "experiments": Object {
    "restapp_header_help_button": Object {
      "enabled": true,
    },
  },
}
`;
